### ISSUE TITLE
Disambiguation to concept.

### ISSUE THREAD

**Issue by BengtGustafsson** [atomgalaxy/isocpp-universal-template-param#38]
Logically, if a concept can bind to a UTP it should be possible to disambiguate the UTP as a concept. Actually I think it is hard to find a use case for concepts as a kind bindable to an UTP, as we (in contrast to D) don't  have if constexpr in class definitions, excluding some methods and adding others for instance. So while it is easy to find a syntactically ok use case for concept template template parameters making them a useful kind for UTPs may be taking things one step too far.

template<template<typename T> concept C> struct S {
    void set(C auto c);
};

Semantically it seems less convincing, what kind of code would set contain that could handle differently constrained c values. Maybe some kind of container which doesn't in itself care about the constraints of its contents, but which could guarantee to users that all specializations of the container for a certain concept has elements modelling the concept:

template<concept C, C T> using constrained_vector = std::vector<T>;

Well, is that anymore useful than a vector itself. Yes maybe, a function could take a constrained_vector<my_concept, T> and rest assured that the elements of the vectors it gets model my_concept. This improves diagnostics. The win is that we don't have to reimplement std::vector for each constraint we want to put on elements.

Note that I misspelled the template parameter as just `concept C` when it should have been `template<typename T> concept C` but on the other hand it seems a reasonable shorthand to allow as there are no non-template concepts anyway.

If this is on the table concept currying would be the next big thing, to allow:

template<typename T> using  SubclassVector = constrained_vector<std::base_of<MyBase>, T>;

Here std::base_of is curried to mimic how concepts with arity > 1 usually are written.

In short it seems that there is much more to concept template parameters than we covered, maybe it should have been its own proposal after all?
